reconcile/rhidp/ocm_oidc_idp/base.py

```diff
@@ -154,13 +162,17 @@ def act(
     ocm_map: OCMMap,
     current_state: Sequence[OCMOidcIdp],
     desired_state: Sequence[OCMOidcIdp],
+    managed_idps: list[str],
 ) -> None:
     """Compare current and desired OIDC identity providers and add, remove, or update them."""
     to_add = set(desired_state) - set(current_state)
     to_remove = set(current_state) - set(desired_state)
     to_compare = set(current_state) & set(desired_state)
 
     for idp in to_remove:
+        if managed_idps and idp.name not in managed_idps:
```

Comment:
If the amount of `managed_idps` is large, consider making it as `set` to accelerate lookup, so far in code I can see it has at most one item, so not a problem for now.